react/ShippingForm.tsx

```diff
@@ -0,0 +1,114 @@
+import { Address } from 'vtex.checkout-graphql'
+import React, { useCallback } from 'react'
+import { OrderForm } from 'vtex.order-manager'
+import { OrderShipping } from 'vtex.order-shipping'
+import { AddressContext } from 'vtex.address-context'
+
+import NewAddressForm from './components/NewAddressForm'
+import AddressList from './components/AddressList'
+import ShippingOptionList from './components/ShippingOptionList'
+import AddressCompletionForm from './components/AddressCompletionForm'
+import ReceiverInfoForm from './components/ReceiverInfoForm'
+import useShippingStateMachine from './shippingStateMachine/useShippingStateMachine'
+
+const ShippingForm: React.FC = () => {
+  const {
+    orderForm: {
+      shipping: { availableAddresses },
+    },
+  } = OrderForm.useOrderForm()
+  const { selectedAddress, deliveryOptions } = OrderShipping.useOrderShipping()
+
+  const { matches, send, state } = useShippingStateMachine({
+    availableAddresses: (availableAddresses as Address[]) ?? [],
+    selectedAddress: selectedAddress ?? null,
+    deliveryOptions,
+  })
+
+  const handleAddressCreated = useCallback(
+    (address: Address) => {
+      send({ type: 'SUBMIT_CREATE_ADDRESS', address })
+    },
+    [send]
+  )
+
+  const handleDeliveryOptionSelect = (deliveryOptionId: string) => {
+    send({ type: 'SUBMIT_SELECT_DELIVERY_OPTION', deliveryOptionId })
+  }
+
+  const handleAddressCompleted = (
+    updatedAddress: Address,
+    buyerIsReceiver: boolean
+  ) => {
+    send({ type: 'SUBMIT_COMPLETE_ADDRESS', buyerIsReceiver, updatedAddress })
+  }
+
+  switch (true) {
+    case matches('completeAddress'): {
```

Comment:
Could those case strings be consts?

Reply:
this parameter is already strongly typed, and TypeScript will throw an error if a typo is present. do you think it would be useful nonetheless?

Reply:
^ this:
<img width="967" alt="Screen Shot 2020-04-28 at 20 07 33" src="https://user-images.githubusercontent.com/10400340/80546409-ed205500-898b-11ea-8e7d-a0b3dbdd4ff2.png">